### PR TITLE
Allow running etcd in v3 API (currently just on GCE)

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #20504 #25286

@timothysc @xiang90 @hongchaodeng @smarterclayton @kubernetes/sig-api-machinery

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30349)

### CODE REVIEW

File: cluster/saltbase/salt/kube-apiserver/kube-apiserver.manifest [hunk 1]
```diff
@@ -54,6 +54,10 @@
   {% set bind_address = "--bind-address=" + grains.publicAddressOverride -%}
 {% endif -%}
 
+{% set storage_backend = "" -%}
+{% if pillar['storage_backend'] is defined -%}
+  {% set storage_backend = "--storage-backend=" + pillar['storage_backend'] -%}
```
Comment (lavalamp) [inline cluster/saltbase/salt/kube-apiserver/kube-apiserver.manifest]: no other parameter changes?

**Comment by lavalamp** [comment]
lgtm

File: cluster/saltbase/salt/kube-apiserver/kube-apiserver.manifest (hunk 1 above)
Comment (wojtek-t) [inline cluster/saltbase/salt/kube-apiserver/kube-apiserver.manifest]: That is my understanding - when I run it in my local project with etcd3 storage backend, it seemed to work (and there weren't any objects under etcd:4001/v2/keys/, which suggests that it was really using v3 API and storage.

Comment (timothysc) [inline cluster/saltbase/salt/kube-apiserver/kube-apiserver.manifest]: it's a port difference, I'm working on the v3client PR right now.

Comment (wojtek-t) [inline cluster/saltbase/salt/kube-apiserver/kube-apiserver.manifest]: Hmm - I'm wondering why it worked for me.

Anyway - waiting for you PR - I'm happy to review it.

**Pull request merged**